yarnitor/background/worker.py

```diff
@@ -278,15 +285,20 @@ def _make_application_handler(self, yarn_application_info):
         -------
         instance of BaseApplicationInfo
         """
-        klass = self.application_handlers.get(yarn_application_info["applicationType"], BaseHandler)
+        app_type = yarn_application_info['applicationType']
+        klass = self.application_handlers.get(app_type, BaseHandler)
         return klass.from_yarn_application_info(yarn_application_info)
 
     def _generate_listing(self):
-        """Computes the listing of applications and the additional information provided by the handlers.
+        """Computes the listing of applications and the additional information
+        provided by the handlers.
 
         """
-        a = self.yarn_handler.cluster_applications("RUNNING")
-        apps = a["apps"]["app"]
+        cluster_apps = self.yarn_handler.cluster_applications("RUNNING")
+        if 'apps' not in cluster_apps or cluster_apps['apps'] is None:
+            logger.warn('No application data available')
+            return {}
+        apps = cluster_apps['apps']['app']
```

Comment:
Yes, and it's meant to bubble and stop processing at the loop level for this iteration.